crates/jstzd/src/task/octez_node.rs

```diff
@@ -0,0 +1,215 @@
+use std::process::Child;
+
+#[derive(Clone)]
+pub struct OctezNodeConfig {
+    /// Path to the octez node binary.
+    binary_path: PathBuf,
+    /// Path to the directory where the node keeps data.
+    data_dir: PathBuf,
+    /// Name of the tezos network that the node instance runs on.
+    network: String,
+    /// HTTP endpoint of the node RPC interface, e.g. 'localhost:8732'
+    rpc_endpoint: String,
+    /// Path to the file that keeps octez node logs.
+    log_file: PathBuf,
+    /// Run options for octez node.
+    options: Vec<String>,
+}
+
+#[derive(Default)]
+pub struct OctezNodeConfigBuilder {
+    binary_path: Option<PathBuf>,
+    data_dir: Option<PathBuf>,
+    network: Option<String>,
+    rpc_endpoint: Option<String>,
+    log_file: Option<PathBuf>,
+    options: Option<Vec<String>>,
+}
+
+impl OctezNodeConfigBuilder {
+    pub fn new() -> Self {
+        OctezNodeConfigBuilder::default()
+    }
+
+    /// Sets the path to the octez node binary.
+    pub fn set_binary_path(&mut self, path: &str) -> &mut Self {
+        self.binary_path = Some(PathBuf::from(path));
+        self
+    }
+
+    /// Sets the path to the directory where the node keeps data.
+    pub fn set_data_dir(&mut self, path: &str) -> &mut Self {
+        self.data_dir = Some(PathBuf::from(path));
+        self
+    }
+
+    /// Sets the name of the tezos network that the node instance runs on.
+    pub fn set_network(&mut self, network: &str) -> &mut Self {
+        self.network = Some(network.to_owned());
+        self
+    }
+
+    /// Sets the HTTP(S) endpoint of the node RPC interface, e.g. 'http://localhost:8732'
+    pub fn set_rpc_endpoint(&mut self, endpoint: &str) -> &mut Self {
+        self.rpc_endpoint = Some(endpoint.to_owned());
+        self
+    }
+
+    /// Sets the path to the file that keeps octez node logs.
+    pub fn set_log_file(&mut self, path: &str) -> &mut Self {
+        self.log_file = Some(PathBuf::from(path));
+        self
+    }
+
+    /// Sets run options for octez node.
+    pub fn set_run_options(&mut self, options: &[&str]) -> &mut Self {
+        self.options = Some(
+            options
+                .iter()
+                .map(|v| (*v).to_owned())
+                .collect::<Vec<String>>(),
+        );
+        self
+    }
+
+    /// Builds a config set based on values collected.
+    pub fn build(&mut self) -> Result<OctezNodeConfig> {
+        Ok(OctezNodeConfig {
+            binary_path: self
+                .binary_path
+                .take()
+                .unwrap_or(PathBuf::from("octez-node")),
+            data_dir: self
+                .data_dir
+                .take()
+                .unwrap_or(PathBuf::from(tempfile::TempDir::new().unwrap().path())),
+            network: self.network.take().unwrap_or("sandbox".to_owned()),
+            rpc_endpoint: self
+                .rpc_endpoint
+                .take()
+                .unwrap_or("localhost:8732".to_owned()),
```

Comment:
Could you extract "localhost:8732" into a `const`?

Reply:
Done. I also moved `sandbox` and `octez-node` into constants.